EIPS/eip-5585.md

```diff
@@ -0,0 +1,141 @@
+    function userLimitUpdate(unit256 userLimit) external onlyOwner;
+  
+    /// @notice resetAllowed flag can be updated by contract owner to control whether the authorization can be revoked or not 
+    /// @param resetAllowed It is the boolean flag
+    function resetAllowedUpdate(bool resetAllowed) external onlyOwner;
+
+
+    /// @notice Check if the token is available for authorization
+    /// @dev Throws if tokenId is not a valid NFT
+    /// @param tokenId The NFT to be checked the availability
+    /// @return true or false whether the NFT is available for authorization or not
+    function authorizationAvailabilityCheck (uint256 tokenId) public view returns(bool);
+
+    /// @notice Clear authorization of a specified user
+    /// @dev The zero address indicates there is no user. The function  works when resetAllowed is true and it will throw exception when false  
+    /// @param tokenId The NFT on which the authorization based
+    /// @param user The user whose authorization will be cleared
+    function resetUser(uint256 tokenId, address user) external;
+
+    /// @notice This is an OPTIONAL function that the operator MAY call, he can set the starting time of staking as a reward of the authorization for each user 
+    /// @dev The zero address indicates there is no user
+    /// @param user To which user the staking time will be set
+    /// @param stakingTime The starting time of the staking for each user
+    function stakingTimeUpdate(address[] user, uint[] stakingTime) external;
+
+
+    /// @notice Emitted when the user of a NFT is changed or the authorization expires time is updated
+    /// param tokenId The NFT on which the authorization based
+    /// param indexed user The user to whom the NFT authorized
+    /// param expires The expires time of the authorization
+    event userAuthorization(uint256 indexed tokenId, address indexed user, uint expires);
+}
+```
+
+
+The `userAuthorization(uint256 tokenId, address user, uint duration)` function MAY be implemented as `public` or `external`.
+
+The `durationUpdate(uint256 tokenId, address user, uint duration)` function MAY be implemented as `public` or `external`.
+
+The `userExpiresTime(uint256 tokenId, address user)` function MAY be implemented as `pure` or `view`.
+
+The `userLimitUpdate(unit256 userLimit)` function MAY be implemented as`public` or `external`.
+
+The `resetAllowedUpdate(bool resetAllowed)` function MAY be implemented as `public` or `external`.
+
+The `authorizationAvailabilityCheck (uint256 tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `resetUser(uint256 tokenId, address user)` function MAY be implemented as `public` or `external`.
+
+The `stakingTimeUpdate(address[] user, uint[] stakingTime)` function MAY be implemented as `public` or `external`.
+
+The `userAuthorization` event MUST be emittedwhen the user of a NFT is changed or the authorization expires time is updated.
+
+
+
+## Rationale
```

Comment:
Please fix the grammar issues with this section.

Reply:
Not a problem, thanks.